examples/mnist_with_tensorboard.py

```diff
@@ -0,0 +1,172 @@
+        self.conv2_drop = nn.Dropout2d()
+        self.fc1 = nn.Linear(320, 50)
+        self.fc2 = nn.Linear(50, 10)
+
+    def forward(self, x):
+        x = F.relu(F.max_pool2d(self.conv1(x), 2))
+        x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
+        x = x.view(-1, 320)
+        x = F.relu(self.fc1(x))
+        x = F.dropout(x, training=self.training)
+        x = self.fc2(x)
+        return F.log_softmax(x)
+
+
+def get_plot_training_loss_handler(writer, plot_every):
+    def plot_training_loss_to_tensorboard(trainer):
+        if trainer.current_iteration % plot_every == 0:
+            writer.add_scalar("training/loss",
+                              trainer.training_history.simple_moving_average(window_size=100),
+                              trainer.current_iteration)
+    return plot_training_loss_to_tensorboard
+
+
+def get_plot_validation_loss_handler(writer):
+    def plot_validation_loss_to_tensorboard(trainer):
+        avg_loss = np.mean([loss for (loss, accuracy) in trainer.validation_history])
+        writer.add_scalar("validation/loss",
+                          avg_loss,
+                          trainer.current_iteration)
+    return plot_validation_loss_to_tensorboard
+
+
+def get_plot_validation_accuracy_handler(writer, validation_data):
+    def plot_val_accuracy_to_tensorboard(trainer):
+        accuracy = sum([accuracy for (loss, accuracy) in trainer.validation_history])
+        accuracy = (accuracy * 100.) / len(validation_data.dataset)
+        writer.add_scalar("validation/accuracy",
+                          accuracy,
+                          trainer.current_iteration)
+    return plot_val_accuracy_to_tensorboard
+
+
+def get_log_validation_loss_and_accuracy_handler(logger, validation_data):
+    def log_validation_loss_and_accuracy(trainer):
+        avg_loss = np.mean([loss for (loss, accuracy) in trainer.validation_history])
+        accuracy = sum([accuracy for (loss, accuracy) in trainer.validation_history])
+        logger('\nValidation set: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n'.format(
+            avg_loss, accuracy, len(validation_data.dataset),
+            (accuracy * 100.) / len(validation_data.dataset)
+        ))
+    return log_validation_loss_and_accuracy
+
+
+def run(batch_size, val_batch_size, epochs, lr, momentum, log_interval, logger, log_dir):
+
+    writer = SummaryWriter(log_dir=log_dir)
+    data_transform = Compose([ToTensor(), Normalize((0.1307,), (0.3081,))])
+
+    train_loader = DataLoader(MNIST(download=True, root=".", transform=data_transform, train=True),
+                              batch_size=batch_size, shuffle=True)
+
+    val_loader = DataLoader(MNIST(download=False, root=".", transform=data_transform, train=False),
+                            batch_size=val_batch_size, shuffle=False)
+
+    model = Net()
+    optimizer = SGD(model.parameters(), lr=lr, momentum=momentum)
+
+    try:
+        writer.add_graph_onnx(model)
```

Comment:
what does this do?

Reply:
This should add a graph of the model using onnx:
See [this](https://github.com/lanpa/tensorboard-pytorch/blob/master/demo_graph.py) and [this](https://github.com/lanpa/tensorboard-pytorch/blob/master/tensorboardX/writer.py#L395)